mws/sitesmanagement/cronjobs.py

```diff
@@ -309,3 +310,67 @@ def reject_or_accepted_old_domain_names_requests():
                                    "%s days.") % (grace_days,))
         else:
             domain_name.accept_it()
+
+@shared_task(base=ScheduledTaskWithFailure)
+def validate_domains():
+    '''
+    Iterate over DomainName objects and set them to:
+     - global if they are visible to (currently) Google's nameservers
+     - private if they are onl`y available to the Cambridge nameservers
+     - deleted if they are visible to none of the above.
+    '''
+    active_states = ['accepted', 'private', 'global', 'deleted']
+
+    for domainname in DomainName.objects.filter(status__in=active_states):
+            status = domainname.validate()
+            if status != domainname.status:
```

Comment:
this is done in validate() already, you should call it with update=true
https://github.com/uisautomation/mws-panel/blob/20795b01473ebcde182f851e5a6b6a3100d8810b/mws/sitesmanagement/models.py#L733